src/inputcard.ts

```diff
@@ -142,40 +148,109 @@ document.addEventListener('DOMContentLoaded', () => {
+            return sessionToken;
           },
-        })
-      ).fold(
-        () => undefined, // to be replaced with logic to handle failures
-        () => {
-          window.location.replace('check.html');
-        },
-      );
-      /*
-      e.preventDefault();
-      const result = await pmClient.startSessionUsingPOST({
-        startSessionRequest: {
-          data: {
-            email: 'pippo@pluto.com',
-            fiscalCode: 'HBBJUU78U89R556T',
-            idPayment: '12345',
+        )
+        .run();
+
+      // debug(`Bearer ${mySessionToken}`);
+      await TE.tryCatch(
+        () =>
+          pmClient.approveTermsUsingPOST({
+            Bearer: `Bearer ${mySessionToken}`,
+            approveTermsRequest: {
+              data: {
+                terms: true,
+                privacy: true,
+              },
+            },
+          }),
+        toError,
+      )
+        .fold(
+          () => undefined, // to be replaced with logic to handle failures
+          myResExt => {
+            const approvalState = myResExt.fold(
+              () => 'noApproval',
+              myRes => (myRes.status === 200 ? JSON.stringify(myRes.value.data) : 'noApproval'),
+            );
+            sessionStorage.setItem('approvalState', approvalState);
           },
-        },
-      });
+        )
+        .run();
+
+      await TE.tryCatch(
+        () =>
+          pmClient.addWalletUsingPOST({
+            Bearer: `Bearer ${mySessionToken}`,
+            walletRequest: {
+              data: {
+                type: TypeEnum.CREDIT_CARD,
+                creditCard: {
+                  expireMonth: (creditcardformExpiration as HTMLInputElement).value.split('/')[0],
+                  expireYear: (creditcardformExpiration as HTMLInputElement).value.split('/')[1],
+                  holder: (creditcardformName as HTMLInputElement).value.trim(),
+                  pan: (creditcardformNumber as HTMLInputElement).value.trim(),
+                  securityCode: (creditcardformSecurecode as HTMLInputElement).value,
+                },
+                idPagamentoFromEC: checkData.idPayment, // needs to exist
+              },
+            },
+            language: 'it',
+          }),
+        toError,
+      )
+        .fold(
+          () => void 0, // to be replaced with logic to handle failures
+          myResExt => {
+            const walletResp = myResExt.fold(
+              () => 'fakeCC',
+              myRes => (myRes.status === 200 ? JSON.stringify(myRes.value.data) : 'fakeWallet'),
+            );
+            sessionStorage.setItem('wallet', walletResp);
+            window.location.replace('check.html');
+          },
+        )
+        .run();
 
-      if (result.isRight()) {
-        creditcardformInputs?.forEach(el => {
-          sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
-        });
-        window.location.replace('check.html');
-      } */
+      /*
```

Comment:
Is it useful ?

Reply:
No, I'll remove it.. thank you for pointing out